godel_surface_detection/src/scan/profilimeter_scan.cpp

```diff
@@ -0,0 +1,237 @@
+#include "godel_surface_detection/scan/profilimeter_scan.h"
+#include <ros/ros.h>
+using namespace godel_process_path;
+
+// Local types
+namespace
+{
+  // stub for gift-wrapping algorithm to compute convex hull
+  // stub for rotating-calipers algorithm to compute min rectangle
+
+  /**
+   * Rotated rectangle structure for use with striping algorithm
+   * (x,y) is the center (of mass)
+   * w, h are the edge lengths
+   * a is the clockwise rotation from upright
+   */
+  struct RotatedRect
+  {
+    double x, y, w, h, a;
+  };
+
+  RotatedRect simpleBoundingBox(const PolygonBoundary& boundary)
+  {
+    double min_x = boundary[0].x;
+    double min_y = boundary[0].y;
+    double max_x = boundary[0].x;
+    double max_y = boundary[0].y;
+
+    for (std::size_t i = 1; i < boundary.size(); ++i)
+    {
+      const PolygonPt& pt = boundary[i];
+      min_x = std::min(pt.x, min_x);
+      min_y = std::min(pt.y, min_y);
+      max_x = std::max(pt.x, max_x);
+      max_y = std::max(pt.y, max_y); 
+    }
+
+    RotatedRect result;
+    result.a = 0.0; // non-rotated, hence the simple bounding box
+    result.x = (max_x + min_x) / 2;
+    result.y = (max_y + min_y) / 2;
+    result.w = max_x - min_x;
+    result.h = max_y - min_y;
+    return result;
+  }
+
+  // Vertically or horizontally?
+  std::vector<RotatedRect> sliceBoundingBox(const RotatedRect& bbox, double slice_width, double overlap)
+  {
+    // we assume slice_width > overlap
+    int n_slices;
+    float dx, dy; // unit vector
+    float x, y; // start position
+    float w, h;
+
+    float adjusted_width = slice_width - overlap;
```

Comment:
Is there any chance that this may produce a negative value? Perhaps it should go inside a std::abs(...)